From 7692089e59d5dbba58e0c5b4f15a1ac21df32e75 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Wed, 8 Aug 2018 15:06:58 +0800
Subject: [PATCH 4407/5242] MLK-19168-3: ASoC: imx-pcm-rpmsg: register rpmsg
 wm8960 and cs42xx8 codec

commit  7ffb08f0c5601d126f369c1d595f7b8326177247 from
https://source.codeaurora.org/external/imx/linux-imx.git

register rpmsg wm8960 and cs42xx8 codec

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
(cherry picked from commit c49f8d20c6fd4479ad45d76290bb5c57d4800d9e)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-pcm-rpmsg.c |   27 ++++++++++++++++++++++++---
 1 file changed, 24 insertions(+), 3 deletions(-)

diff --git a/sound/soc/fsl/imx-pcm-rpmsg.c b/sound/soc/fsl/imx-pcm-rpmsg.c
index 5a46e1f..76a6fa6 100644
--- a/sound/soc/fsl/imx-pcm-rpmsg.c
+++ b/sound/soc/fsl/imx-pcm-rpmsg.c
@@ -708,6 +708,7 @@ static int i2s_rpmsg_probe(struct rpmsg_device *rpdev)
 {
 	struct platform_device *codec_pdev;
 	struct fsl_rpmsg_i2s *rpmsg_i2s = NULL;
+	struct fsl_rpmsg_codec  rpmsg_codec[2];
 	int ret;
 
 	if (!i2s_info_g)
@@ -722,12 +723,15 @@ static int i2s_rpmsg_probe(struct rpmsg_device *rpdev)
 
 	rpmsg_i2s = container_of(i2s_info_g, struct fsl_rpmsg_i2s, i2s_info);
 
-	if (rpmsg_i2s->codec) {
+	if (rpmsg_i2s->codec_wm8960) {
+		rpmsg_codec[0].audioindex = rpmsg_i2s->codec_wm8960 >> 16;
+		rpmsg_codec[0].shared_lrclk = true;
+		rpmsg_codec[0].capless = false;
 		codec_pdev = platform_device_register_data(
 					&rpmsg_i2s->pdev->dev,
-					RPMSG_CODEC_DRV_NAME,
+					RPMSG_CODEC_DRV_NAME_WM8960,
 					PLATFORM_DEVID_NONE,
-					NULL, 0);
+					&rpmsg_codec[0], sizeof(struct fsl_rpmsg_codec));
 		if (IS_ERR(codec_pdev)) {
 			dev_err(&rpdev->dev,
 				"failed to register rpmsg audio codec\n");
@@ -736,6 +740,23 @@ static int i2s_rpmsg_probe(struct rpmsg_device *rpdev)
 		}
 	}
 
+	if (rpmsg_i2s->codec_cs42888) {
+		rpmsg_codec[1].audioindex = rpmsg_i2s->codec_cs42888 >> 16;
+		strcpy(rpmsg_codec[1].name, "cs42888");
+		rpmsg_codec[1].num_adcs = 2;
+
+		codec_pdev = platform_device_register_data(
+					&rpmsg_i2s->pdev->dev,
+					RPMSG_CODEC_DRV_NAME_CS42888,
+					PLATFORM_DEVID_NONE,
+					&rpmsg_codec[1], sizeof(struct fsl_rpmsg_codec));
+		if (IS_ERR(codec_pdev)) {
+			dev_err(&rpdev->dev,
+				"failed to register rpmsg audio codec\n");
+			ret = PTR_ERR(codec_pdev);
+			return ret;
+		}
+	}
 	return 0;
 }
 
-- 
1.7.9.5

